src/citrine/_serialization/properties.py

```diff
@@ -696,39 +696,71 @@ def __str__(self):
 
 
 class Mapping(Property[dict, dict]):
+    """
+    Serialization of a Mapping.
+
+    Serialization is done by converting the map to a dict by default, serializing the
+    keys and the values.
+
+    If the optional parameter `ser_as_list_of_pairs` is set to True, serialization is done
+    by converting the map to a list of key value pairs. Deserialization expects a list of
+    key value pairs and converts them to a dict.
+    """
+
     def __init__(self,
                  keys_type: typing.Union[Property, typing.Type[Property]],
                  values_type: typing.Union[Property, typing.Type[Property]],
                  serialization_path: typing.Optional[str] = None,
                  serializable: bool = True,
                  deserializable: bool = True,
                  default: typing.Optional[dict] = None,
-                 override: bool = False):
+                 override: bool = False,
+                 ser_as_list_of_pairs: bool = False):
         super().__init__(serialization_path,
                          serializable,
                          deserializable,
                          default,
                          override)
         self.keys_type = keys_type if isinstance(keys_type, Property) else keys_type()
         self.values_type = values_type if isinstance(values_type, Property) else values_type()
+        self.ser_as_list_of_pairs = ser_as_list_of_pairs
 
     @property
     def underlying_types(self):
         return dict
 
     @property
     def serialized_types(self):
-        return dict
+        if self.ser_as_list_of_pairs:
+            return list
+        else:
+            return dict
 
-    def _deserialize(self, value: dict) -> dict:
+    def _deserialize(self, value: typing.Union[dict, list]) -> dict:
         deserialized = dict()
+
+        if self.ser_as_list_of_pairs:
```

Comment:
This logic would result in an error if `self.ser_as_list_of_pairs` and the `value` is a `dict`.  I'm not sure that's the behavior we want: it might be preferable to always be able to deserialize either format and have `ser_as_list_of_pairs` only affect serialization logic.  That would make forward and backward compatibility eaiser.

Also, since list and dict are distinct, we can just do `if type(value) == list`

Reply:
yeah, that's what I had originally. I changed it to this to constrain it more specifically so that any errors in serialization were caught earlier. Mostly, I was thinking that if the python client is sending the data serialized as a list and the backend is correctly deserializing that, then the backend should also be sending the data serialized as a list. And if it is not, then something is wrong.

But I think that you are right that it will make backwards compatible changes somewhat easier if we can handle both in this client, and will change it back.

Reply:
The behavior could also be parameterized, if we want to open the backward compatibility window and close it later to be more strict.